src/deadline/job_attachments/download.py

```diff
@@ -490,6 +516,18 @@ def process_client_error(exc: ClientError, status_code: int):
         # TODO: Temporary to prevent breaking backwards-compatibility; if file not found, try again without hash alg postfix
         status_code = int(exc.response["ResponseMetadata"]["HTTPStatusCode"])
         if status_code == 404:
+
+            def handler(bytes_downloaded):
+                nonlocal progress_tracker
+                nonlocal future
+
+                if progress_tracker:
+                    should_continue = progress_tracker.track_progress_callback(bytes_downloaded)
+                    if not should_continue:
+                        future.cancel()
+
+            subscribers = [ProgressCallbackInvoker(handler)]
```

Comment:
What's this used for, maybe it can use some documentation too?

Reply:
I refactored this part form the original `download_file` command, which just provides statistics for the downloading progress

Reply:
I've done some extensive testing and reviewing:
- ran existing integ tests
- ran manual end to end testing by downloading outputs, cancelling downloaded outputs, and comparing these outputs with the same process on mainline
- consulted previous implementer of this part of code, which seemed good

I don't believe there is any regression with this refactoring.